src/main/riddl/Members/members.riddl

```diff
@@ -146,30 +147,14 @@ context MembersContext  is {
       |that the member is now active.
     }
 
-    command InactivateMember is {memberId: MemberId, actingMember: MemberId}
-    briefly "Update state of member to Inactive"
-    described as {
-      |This command inactivates the member.
-    }
-
-    event MemberInactivated is {memberId: MemberId, meta: MembersContext.MetaInfo}
-    briefly "Event issued after InactivateMember"
-    described as {
-      |This event is issued after InactivateMember to denote
-      |that the member is now inactive.
+    command UpdateMemberStatus is {
+      memberId: MemberId,
+      status: ^MemberContext.StatusUpdates,
+      updatingMember: MemberId
     }
```

Comment:
I actually prefer the previous arrangement. I would like to see a set of three commands, and corresponding events, for each of the "StatusUpdates" (MemberStates) enumerators.  I am going to object to any command that starts with "Update" or "Change".  They are are lazy verbs, a lot like "nice" and "good" are lazy adjectives.  They don't provide enough information.  For example, `InactivateMember` leads one to understand that a member is going to become inactive, just on the name alone, while `UpdateMemberStatus` doesn't give you as much information because you don't know what the member's status will be without looking at the RUNTIME value of a field of field that is not discernable at time of reading the code.  APIs need to be intuitive too.

Reply:
I can change back but don't really understand the reasoning here. `UpdateMemberStatus` will always contain what the member's status will be in the `status` field. A command with a field that says the Status type doesn't seem much worse than a command named with that Status type. It would be probably easier to pair results/errors with commands though.

Reply:
There is no way to determine the status implied by the UpdateMemberStatus definition. All you know is that the status will change (perhaps), but you don't know what it will become. Inspection of its definition of that field still doesn't tell you the ACTUAL value that will be set, only that it is of "Status" type. ActivateMember is very clear. You don't even have to look at the content of the message.  What if, later on, you wanted to change the representation of the status from an enum to a set of records (not RIDDL states). You would have to change UpdateMemberStatus to expose internal details of the entity. That change could have a huge ripple effect on all the users of Member.  With ActivateMember you don't have to do this and it has no ripple effect. It is just an encapsulated implementation detail.  APIs need to be designed defensively, just like driving a car.  If you still don't get it, please just trust me. This is the right way to design an API.

Reply:
Ok, that makes total sense, being cautious and thinking ahead :)